extensions/resource-deployment/src/services/tools/kubeCtlTool.ts

```diff
@@ -50,8 +49,7 @@ export class KubeCtlTool extends ToolBase {
 		let version: SemVer | undefined = undefined;
 		if (output) {
 			const versionJson: KubeCtlVersion = JSON.parse(output);
-			// kubectl version output might contain '+' character in the minor version, e.g. 16+, we have to remove it to make it a valid semantic version string.
-			version = new SemVer(`${versionJson.clientVersion.major}.${versionJson.clientVersion.minor.replace(/\+/g, '')}.0`);
+			version = new SemVer(versionJson.clientVersion.gitVersion);
```

Comment:
Given that we're parsing raw JSON you should be checking that all the properties are defined as expected - throwing a useful error if not.

Reply:
It is extremely unlikely that the output of Kubectl command-line execution will be missing the clientVersion.gitVersion fields in the emitted output, so alternately you can use null coalescence to return some sentinel values in those scenarios. Since the whole output produced from kubeCtl is already logged to the output channel, it would be pretty easy to debug when the sentinel value shows up as the KubeCtl version in the tools table.